os/fs/smartfs/smartfs_utils.c

```diff
@@ -1223,13 +1231,11 @@ int smartfs_find_availableentry(struct smartfs_mountpt_s *fs, uint16_t *parentdi
 		psector = nextsector;
 	}
 
-	*parentdirsector = direntry->dsector;
+	direntry->prev_parent = direntry->dsector;
+	/* No available entry was found, so we create one */
+	ret = smartfs_createdirentry(fs, direntry);
 	if (ret != OK) {
-		/* No available entry was found, so we create one */
-		ret = smartfs_createentry(fs, *parentdirsector, direntry);
-		if (ret != OK) {
-			fdbg("Createentry unable to create space for writing\n");
-		}
+		fdbg("Createentry unable to create space for writing\n");
```

Comment:
it is very minor one, let's remove 'if' in this commit.

Reply:
You mean Line 1237 to check if createdirentry returns OK or not?